src/main/g8/default.properties

```diff
@@ -14,5 +14,7 @@ http4s_version = 0.21.3
 circe_version = 0.13.0
 logback_version = 1.2.3
 specs2_version = 4.9.3
+# graal_vm_specific
 graal_native_image = true
+is_linux_build = false 
```

Comment:
I would imagine it's common to run the same source on both Mac and Linux.

Maybe instead of being fancy about conditional logic in the README, we could create a launcher shell script.  And that could check the running version, and do the right thing out of the box on MacOS and Linux.

Reply:
There's not terribly much difference between that approach and what we've provided. Since it was already suggested to use conditionals in the G8 template I went ahead and set the parameters there.

Another thing to mention. The documentation and template have taken the approach regarding educating the user. If they run into an issue building a native-image, I'd like them to ideally help troubleshoot it and solve it themselves. I am happy about them launching a few commands and installing a few things themselves. If we did that all for them when something breaks they might have no idea how to fix it, and even the ability to sort out what was going on.

I could write a launcher script. I'm open to suggestions regarding doing that. Do you have any examples of detecting mac / windows / linux that you find elegant? Finally please see the comment above about writing well defined issues and possibly a brainstorming meetup or chat.

Reply:
Okay, that's a good point on education with the readme vs. a script.  I still think the graal_native_image conditional is more compelling than the is_linux conditional: the same service won't often be deployed with and without Graal, while it will often be run on multiple OSes.  (My last several teams have been mixed OS.)  My fear is bombarding them with too many questions.  I'd probably just document it both ways in the readme.

If you want to also add a launcher script, or just for future reference, the `$OSTYPE` environment variable should usually work.  `linux-gnu` for Linux, `darwin` for MacOS.

Reply:
Hi @rossabaker , we are already parameterizing and configuring the readme via gitter8 properties not only via `graal_native_image`, but also by the assembly name. If we were to "mix in" shell for some configuration things are getting pretty tacky. Furthermore I went ahead and tried to create the shell script via the giter8 template. But the template language is getting confused by the shell syntax and the docs don't even cover that. It should be easy to escape right?

To some degree I can understand why people would want to use an sbt wrapper to do everything. I wa s convinced that it would be more valuable to get people to learn the native image tooling than trigger commands and use sbt plugins. But I'm not crazy about using obscure templating languages to write shell scripts...

 Can we merge the improvement now and then maybe discuss what the next best steps are moving forward? I have some ideas on other things to do.